mailjet.install

```diff
@@ -1,83 +1,78 @@
-    $errorData = array(
-      'severity' => REQUIREMENT_ERROR,
-      'description' => $t('Mailjet module requires the <a href="@phpmailer">PHPMailer Library</a>, which is missing. Download and extract the entire contents of the archive into the %path directory on your server.',
-        array(
-          '@phpmailer' => 'http://github.com/PHPMailer/PHPMailer/archive/v5.2.21.zip',
-          '%path' => 'sites/all/libraries/phpmailer'
-        )
-      ),
-    );
-
-    if (function_exists('libraries_get_path')) {
-      $phpmailer_path = libraries_get_path('phpmailer');
-      if (!$phpmailer_path) {
-        $requirements['phpmailer'] = $errorData;
-      }
-    }
-    elseif (!file_exists('sites/all/libraries/phpmailer/PHPMailerAutoload.php')) {
-      $requirements['phpmailer'] = $errorData;
-    }
-  }
-  return $requirements;
-}
+<?php
+
+/**
+ * @file
+ * Functions called while installing, uninstalling or activation of the module.
+ */
+
+/**
+ * Implements hook_install().
+ */
+function mailjet_install() {
+  variable_set('mailjet_debug', FALSE);
+  cache_clear_all();
+}
+
+/**
+ * Implements hook_uninstall().
+ */
+function mailjet_uninstall() {
+  variable_del('mailjet_from');
+  variable_del('mailjet_host');
+  variable_del('mailjet_on');
+  variable_del('mailjet_password');
+  variable_del('mailjet_port');
+  variable_del('mailjet_protocol');
+  variable_del('mailjet_test');
+  variable_del('mailjet_test_address');
+  variable_del('mailjet_username');
+  variable_del('mailjet_allowhtml');
+  variable_del('mailjet_debug');
+}
+
+/**
+ * Implements hook_disable().
+ */
+function mailjet_disable() {
+  variable_set('mailjet_on', FALSE);
+  variable_set('mail_system', array('default-system' => 'DefaultMailSystem'));
+}
+
+/**
+ * Implements hook_update().
+ */
+function mailjet_update_7000() {
+  if (variable_get('mailjet_on', FALSE)) {
+    variable_set('mail_system', array('default-system' => 'MailjetSmtpMailSystem'));
+  }
+  cache_clear_all();
+}
+
+/**
+ * Implements hook_requirements().
+ */
+function mailjet_requirements($phase) {
+  $requirements = array();
+  $t = get_t();
+  if ($phase == 'install') {
+    $error_data = array(
+      'severity' => REQUIREMENT_ERROR,
+      'description' => $t('Mailjet module requires the <a href="@phpmailer">PHPMailer Library</a>, which is missing. Download and extract the entire contents of the archive into the %path directory on your server.', array(
+        '@phpmailer' => 'http://github.com/PHPMailer/PHPMailer/archive/v5.2.21.zip',
+        '%path' => 'sites/all/libraries/phpmailer'
+          )
+      ),
+    );
+
+    if (function_exists('libraries_get_path')) {
+      $phpmailer_path = libraries_get_path('phpmailer');
+      if (!$phpmailer_path) {
+        $requirements['phpmailer'] = $error_data;
+      }
+    }
+    elseif (!file_exists('sites/all/libraries/phpmailer/PHPMailerAutoload.php')) {
+      $requirements['phpmailer'] = $error_data;
+    }
+  }
+  return $requirements;
+}
```

Comment:
All text files should end in a single newline (\n). This should be applied to all modified files.